py-polars/polars/lazyframe/frame.py

```diff
@@ -3300,7 +3302,11 @@ def with_context(self, other: Self | list[Self]) -> Self:
 
         return self._from_pyldf(self._ldf.with_context([lf._ldf for lf in other]))
 
-    def drop(self, columns: str | Collection[str], *more_columns: str) -> Self:
+    def drop(
+        self,
+        columns: str | Collection[str] | SelectorType,
+        *more_columns: str | SelectorType,
```

Comment:
It should technically be this, right? 

```suggestion
        columns: str | SelectorType | Collection[str | SelectorType],
```

Same for other `columns` parameters. So I would advise making a type alias for this combination:

`ColumnType: TypeAlias = Union[str, SelectorType]`

Then all type signatures become a lot clearer, like:

```python
        columns: ColumnType | Collection[ColumnType],
        *more_columns: ColumnType,
```

Reply:
Yup, agreed! Though will probably go with the more explicit `ColumnNameOrSelector` for this 🤔

Reply:
Yea that sounds like a better name! I was just illustrating the idea.

Reply:
Quite right :)

Have added two, since we need to distinguish collection/sequence variants:
* `ColumnNameOrSelectorCollection`
* `ColumnNameOrSelectorSequence`

Reply:
Done/committed :)

Reply:
Assume I just face-palmed myself for misunderstanding; now you spelled it out I see _exactly_ what you mean, and it's clearly what I should have done in the first place! These aren't nitpicks, they are constructive criticism...🤣

Reply:
Fixed!